rdk/rdk.py

```diff
@@ -1337,6 +1552,340 @@ def deploy(self):
+                        'ParameterKey': 'Description',
+                        'ParameterValue': rule_description,
+                    },
+                    {
+                        'ParameterKey': 'SourceEvents',
+                        'ParameterValue': source_events,
+                    },
+                    {
+                        'ParameterKey': 'SourcePeriodic',
+                        'ParameterValue': source_periodic,
+                    },
+                    {
+                        'ParameterKey': 'SourceInputParameters',
+                        'ParameterValue': json.dumps(combined_input_parameters),
+                    },
+                    {
+                        'ParameterKey': 'SourceIdentifier',
+                        'ParameterValue': rule_params['SourceIdentifier']
+                    }]
+                my_cfn = my_session.client('cloudformation')
+
+                #deploy config rule
+                cfn_body = os.path.join(path.dirname(__file__), 'template',  "configManagedRuleOrganization.json")
+
+                try:
+                    my_stack_name = self.__get_stack_name_from_rule_name(rule_name)
+                    my_stack = my_cfn.describe_stacks(StackName=my_stack_name)
+                    #If we've gotten here, stack exists and we should update it.
+                    print ("Updating CloudFormation Stack for " + rule_name)
+                    try:
+                        cfn_args = {
+                            'StackName': my_stack_name,
+                            'TemplateBody': open(cfn_body, "r").read(),
+                            'Parameters': my_params
+                        }
+
+                        # If no tags key is specified, or if the tags dict is empty
+                        if cfn_tags is not None:
+                            cfn_args['Tags'] = cfn_tags
+
+                        response = my_cfn.update_stack(**cfn_args)
+                    except ClientError as e:
+                        if e.response['Error']['Code'] == 'ValidationError':
+                            if 'No updates are to be performed.' in str(e):
+                                #No changes made to Config rule definition, so CloudFormation won't do anything.
+                                print("No changes to Config Rule.")
+                            else:
+                                #Something unexpected has gone wrong.  Emit an error and bail.
+                                print(e)
+                                return 1
+                        else:
+                            raise
+                except ClientError as e:
+                    #If we're in the exception, the stack does not exist and we should create it.
+                    print ("Creating CloudFormation Stack for " + rule_name)
+                    cfn_args = {
+                        'StackName': my_stack_name,
+                        'TemplateBody': open(cfn_body, "r").read(),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *open*. The resource is allocated by call *builtins.open*. Currently, there are execution paths that do not contain closure statements, for example, when *builtins.print()* throws an exception, and an if condition at line 1697 evaluates to true. To prevent this resource leak, close the object returned by *open()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).